gtime/forecasting/simple_models.py

```diff
@@ -0,0 +1,354 @@
+    >>> from gtime.forecasting import SeasonalNaiveModel
+    >>> idx = pd.period_range(start='2011-01-01', end='2012-01-01')
+    >>> np.random.seed(1)
+    >>> df = pd.DataFrame(np.random.random((len(idx), 1)), index=idx, columns=['1'])
+    >>> y = horizon_shift(df, horizon=5)
+    >>> X_train, y_train, X_test, y_test = FeatureSplitter().transform(df, y)
+    >>> m = SeasonalNaiveModel(seasonal_length=3)
+    >>> m.fit(X_train, y_train).predict(X_test)
+                         y_1       y_2       y_3       y_4       y_5
+        2011-12-28  0.990472  0.300248  0.782749  0.990472  0.300248
+        2011-12-29  0.300248  0.782749  0.990472  0.300248  0.782749
+        2011-12-30  0.782749  0.990472  0.300248  0.782749  0.990472
+        2011-12-31  0.990472  0.300248  0.782749  0.990472  0.300248
+        2012-01-01  0.300248  0.782749  0.990472  0.300248  0.782749
+    """
+
+
+    def __init__(self, seasonal_length: int):
+
+        super().__init__()
+        self.seasonal_length = seasonal_length
+
+    def fit(self, X: pd.DataFrame, y=None):
+        """Stores the seasonal pattern from the last ``self.seasonal_length`` observations
+
+        Parameters
+        ----------
+        X : pd.DataFrame, shape (n_samples, n_features), train sample, required for compatibility, not used for a naive model.
+
+        y : None
+            Used to store the predict feature names and prediction horizon.
+
+        Returns
+        -------
+        self : SeasonalNaiveModel
+            Returns self.
+
+        """
+
+        x_freq = X.index.freq
+        seasonal_td = pd.Timedelta(self.seasonal_length * x_freq.n, x_freq.name)
+        self.season_ = X.loc[X.index.max()-seasonal_td:]
+        self.season_ = self.season_.iloc[-self.seasonal_length:] # TODO think of a better way to get a non-inclusive index
+        super().fit(X, y)
+
+        return self
+
+    def predict(self, X: pd.DataFrame) -> pd.DataFrame:
+
+        """Using stored ``self._season`` predicts a seasonal time series starting from ``X``
+
+        Parameters
+        ----------
+        X: pd.DataFrame, shape (n_samples, 1), required
+            The time series on which to predict.
+
+        Returns
+        -------
+        predictions : pd.DataFrame, shape (n_samples, self._horizon)
+            The output predictions.
+
+        Raises
+        ------
+        NotFittedError
+            Raised if the model is not fitted yet.
+
+        """
+
+        check_is_fitted(self)
+        time_diff = X.index.to_timestamp() - self.season_.index.max().to_timestamp()
+        len_s = len(self.season_)
+        seasonal_pos = time_diff.days.values % len_s
+        y_pred = np.squeeze([self._season_roll_(x, self._horizon_) for x in seasonal_pos], axis=2)
+        predictions = pd.DataFrame(data=y_pred, index=X.index, columns=self._y_columns_)
+
+        return predictions
+
+    def _season_roll_(self, start: int, horizon: int) -> np.array:
```

Comment:
Call it `_season_roll` without underscore at the end

Reply:
fixed